include/zendoo_mc.h

```diff
@@ -559,22 +573,25 @@ extern "C" {
     typedef struct sc_proof sc_proof_t;
 
     /*
-     * Serialize a proof given an opaque pointer `sc_proof` to it.
-     * Instantiate and return a BufferWithSize containing the proof bytes.
+     * Serialize a proof, in compressed or uncompressed form, depending on the value of `compressed` flag,
+     * given an opaque pointer `sc_proof` to it. Instantiate and return a BufferWithSize containing the proof bytes.
      */
     BufferWithSize* zendoo_serialize_sc_proof(
         const sc_proof_t* proof,
-        CctpErrorCode* ret_code
+        CctpErrorCode* ret_code,
+        bool compressed = true
     );
 
     /*
-     * Deserialize a proof from `sc_proof_bytes` and return an opaque pointer to it.
+     * Deserialize a proof, in compressed or uncompressed form, depending on the value of `compressed` flag,
+     * from `sc_proof_bytes` and return an opaque pointer to it.
      * If `semantic_checks` flag is set, semantic checks on the proof will be performed.
      */
     sc_proof_t* zendoo_deserialize_sc_proof(
         const BufferWithSize* sc_proof_bytes,
         bool semantic_checks,
-        CctpErrorCode* ret_code
+        CctpErrorCode* ret_code,
+        bool compressed = true
```

Comment:
As a further improvement, could we get rid of the `compressed` flag when deserializing? Can it be deduced from the serialized bytes themselves?

Reply:
Unfortunately I don't think so. We would need to know a priori the exact compressed and uncompressed size of the particular instance, and this is not possible (at least not for every type)

Reply:
Isn't it possible to use an approach similar to `zendoo_get_sc_proof_proving_system_type_from_file()`?

Reply:
The proof, pk and vk size is not fixed and can change, regardless of the type

Reply:
Then, what about prepending a byte to the serialized file to specify if it is compressed or not (immediately after or before the _type_ byte)?